### ISSUE TITLE
aws_launch_template add support for Instance Metadata Options

### ISSUE THREAD

**Issue by max-rocket-internet** [terraform-providers/terraform-provider-aws#11794]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently the `aws_launch_template` resource does not support Instance Metadata Options. These options can be required when using the Instance Metadata Service Version 2, which new versions of the AWS SDK now use: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

When calling the [CreateLaunchTemplate](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateLaunchTemplate.html) action in the EC2 API, the [LaunchTemplateData](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestLaunchTemplateData.html) object has a `MetadataOptions` attribute: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateInstanceMetadataOptionsRequest.html

### New or Affected Resource(s)

* aws_launch_template

### Potential Terraform Configuration

Perhaps in terraform it could look like this:

```hcl
resource "aws_launch_template" "foo" {
  name = "foo"
  instance_metadata_options {
    http_endpoint               = "enabled"
    http_put_response_hop_limit = 2
    http_tokens                 = "required"
  }
}
```

### References

https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateInstanceMetadataOptionsRequest.html

**Comment by ewbankkit** [comment]
I think this is the same issue as https://github.com/terraform-providers/terraform-provider-aws/issues/10949.

**Comment by max-rocket-internet** [comment]
@ewbankkit no that's a different resource, the `aws_instance`. This resource is for the `aws_launch_template` which is for instances that are created from an autoscaling group. It's a different AWS API.

**Comment by max-rocket-internet** [comment]
If we can't set this setting, then the result for EKS users can be this: https://medium.com/@ngchiwang/aws-eks-pod-role-5e5554fee501

**Comment by ewbankkit** [comment]
@max-rocket-internet Correct, related but separate.

**Comment by bflad** [comment]
Support for configuring metadata options in the `aws_instance` and `aws_launch_template` resources has been merged and will release with version 2.55.0 of the Terraform AWS Provider, later today. Thanks to @stefansundin and @ewbankkit for the implementation. 👍
